src/Symfony/Component/CssSelector/Parser/Parser.php

```diff
@@ -216,6 +217,21 @@ private function parseSimpleSelector(TokenStream $stream, bool $insideNegation =
                     }
 
                     $result = new Node\NegationNode($result, $argument);
+                } elseif ('has' === strtolower($identifier)) {
+                    if ($insideRelation) {
+                        throw SyntaxErrorException::nestedHas();
+                    }
+
+                    $arguments[] = $this->parserSelectorNode($stream, true);
```

Comment:
Indeed, I changed the way the code parse selectors as python library do. Also added relevant test. 

But now, It looks like the parser can't process descendant relation anymore like `div:has(.foo .bar)`. It would throw an error. I'm pretty sure python one neither do. Should we let it like this for now?